Validating an Access Token with OIDC gives NPE when running in prod mode for the default tenant only

### Discussed in https://github.com/quarkusio/quarkus/discussions/47525

<div type='discussions-op-text'>

<sup>Originally posted by **dometec** April 24, 2025</sup>
Hi Team, I have an Access Token and I want to verify it manually. I use TenantIdentityProvider as written in the [documentation](https://quarkus.io/guides/security-oidc-bearer-token-authentication#authentication-after-an-http-request-has-completed).

The problem is that, when I run the code with quarkud:dev, it works, otherwise I get a NPE:

```
Caused by: java.lang.NullPointerException: Cannot invoke "io.quarkus.oidc.runtime.OidcConfig.namedTenants()" because "config" is null
        at io.quarkus.oidc.runtime.OidcConfig.getDefaultTenant(OidcConfig.java:71)
        at io.quarkus.oidc.runtime.OidcRecorder$TenantSpecificOidcIdentityProvider.<init>(OidcRecorder.java:149)
        at io.quarkus.oidc.runtime.OidcRecorder$4.get(OidcRecorder.java:132)
```

I create a very simple project that demonstrates the problem: https://github.com/dometec/quarkus-iodc-test however all the code is shown below:
```
package testiodc;

import java.time.Duration;

import io.quarkus.oidc.AccessTokenCredential;
import io.quarkus.oidc.TenantIdentityProvider;
import io.quarkus.runtime.StartupEvent;
import io.quarkus.security.identity.SecurityIdentity;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.event.Observes;
import jakarta.inject.Inject;

@ApplicationScoped
public class ApplicationLifecycleListener {

	@Inject
	TenantIdentityProvider identityProvider;

	public void init(@Observes StartupEvent ev) {

		AccessTokenCredential tokenCredential = new AccessTokenCredential("fakeaccesstoken");
		SecurityIdentity securityIdentity = identityProvider.authenticate(tokenCredential).await()
				.atMost(Duration.ofSeconds(30));

		System.out.println(securityIdentity);

	}

}

```

Any suggestion?</div>

--- a/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java
+++ b/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java
@@ -1,7 +1,5 @@
 package io.quarkus.oidc.runtime;
 
-import static io.quarkus.oidc.runtime.OidcConfig.getDefaultTenant;
-import static io.quarkus.oidc.runtime.OidcUtils.DEFAULT_TENANT_ID;
 import static io.quarkus.runtime.configuration.DurationConverter.parseDuration;
 import static io.quarkus.vertx.http.runtime.security.HttpSecurityUtils.getRoutingContextAttribute;
 
@@ -186,12 +184,7 @@ public class OidcRecorder {
             super(Arc.container().instance(DefaultTenantConfigResolver.class).get(),
                     Arc.container().instance(BlockingSecurityExecutor.class).get());
             this.blockingExecutor = Arc.container().instance(BlockingSecurityExecutor.class).get();
-            if (tenantId.equals(DEFAULT_TENANT_ID)) {
-                OidcConfig config = Arc.container().instance(OidcConfig.class).get();
-                this.tenantId = getDefaultTenant(config).tenantId().orElse(OidcUtils.DEFAULT_TENANT_ID);
-            } else {
-                this.tenantId = tenantId;
-            }
+            this.tenantId = tenantId;
         }
 
         @Override

